framework/include/postprocessors/ElementIntegralFunctorPostprocessor.h

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementIntegralPostprocessor.h"
+
+// Forward Declarations
+template <bool>
+class ElementIntegralFunctorPostprocessorTempl;
+typedef ElementIntegralFunctorPostprocessorTempl<false> ElementIntegralFunctorPostprocessor;
+typedef ElementIntegralFunctorPostprocessorTempl<true> ADElementIntegralFunctorPostprocessor;
```

Comment:
You can also just put these after the class definition and then you don't need the forward declaration of the template

Reply:
done